cmd/iam.go

```diff
@@ -469,27 +468,23 @@ func (sys *IAMSys) InfoPolicy(policyName string) ([]byte, error) {
 	if !ok {
 		return nil, errNoSuchPolicy
 	}
-	return json.Marshal(v)
+	return &v, nil
 }
 
 // ListPolicies - lists all canned policies.
-func (sys *IAMSys) ListPolicies() (map[string][]byte, error) {
+func (sys *IAMSys) ListPolicies() (map[string]*iampolicy.Policy, error) {
 	objectAPI := newObjectLayerWithoutSafeModeFn()
 	if objectAPI == nil {
 		return nil, errServerNotInitialized
 	}
 
-	var policyDocsMap = make(map[string][]byte)
-
 	sys.RLock()
 	defer sys.RUnlock()
 
+	policyDocsMap := make(map[string]*iampolicy.Policy, len(sys.iamPolicyDocsMap))
+
 	for k, v := range sys.iamPolicyDocsMap {
-		data, err := json.Marshal(v)
-		if err != nil {
-			return nil, err
-		}
-		policyDocsMap[k] = data
+		policyDocsMap[k] = &v
```

Comment:
This is not correct, &v is always the same in each iteration, what we can do is:
```
p := v
policyDocsMap[k] = &p
```

but also I think we can avoid returning a map of pointers, I am not sure why we are doing this